app/controllers/businesses_controller.rb

```diff
@@ -1,17 +1,18 @@
 class BusinessesController < ApplicationController
+
   def new
     @business = Business.new
   end
 
   def show
-    @business = Business.find(params[:id])
+    @business = Business.find_by(slug: params[:slug]) or render_404
```

Comment:
find by slug please note

as an alternative to overriding to_param within the model is ok here